init/lmod_bash_completions

```diff
@@ -1,168 +1,233 @@
+	if ${redirect,,}; then
+		@PKG@/libexec/lmod bash -t list 2>/dev/null | bash | sed ' /^ *$/d; /:$/d; s#/*$##g; s|^|-|g;'
+	else
+		@PKG@/libexec/lmod bash -t list 2>&1 >/dev/null | sed ' /^ *$/d; /:$/d; s#/*$##g; s|^|-|g;'
+	fi
+}
+
+_module_mcc() {
+	local redirect="${LMOD_REDIRECT:-false}"
+	if ${redirect,,}; then
+		@PKG@/libexec/lmod bash -t collection 2>/dev/null | bash
+	else
+		@PKG@/libexec/lmod bash -t collection 2>&1 >/dev/null
+	fi
 }
 
 _module_not_yet_loaded() {
 	comm -23  <(_module_avail|sort)  <(_module_loaded_modules|sort)
 }
 
 _module_long_arg_list() {
-	local cur="$1" i
-
+	local cur="${1}" i
+	
 	if [[ ${COMP_WORDS[COMP_CWORD-2]} == sw* ]]; then
-		COMPREPLY=( $(compgen -W "$(_module_not_yet_loaded)" -- "$cur") )
+		COMPREPLY=( $(compgen -W "$(_module_not_yet_loaded)" -- "${cur}") )
 		return
 	fi
 	for ((i = COMP_CWORD - 1; i > 0; i--)); do
-	  case ${COMP_WORDS[$i]} in
-	    add|load)
-		COMPREPLY=( $(compgen -W "$(_module_not_yet_loaded)" -- "$cur") )
-		break;;
-	    rm|remove|unload|switch|swap)
-		COMPREPLY=( $(compgen -W "$(_module_loaded_modules)" -- "$cur") )
-		break;;
-	  esac
+		case ${COMP_WORDS[${i}]} in
+		add|load)
+			COMPREPLY=( $(compgen -W "$(_module_not_yet_loaded)" -- "${cur}") )
+			break
+			;;
+		rm|remove|unload|switch|swap)
+			COMPREPLY=( $(compgen -W "$(_module_loaded_modules)" -- "${cur}") )
+			break
+			;;
+		esac
 	done
 }
 
-_module_mcc() {
-	@PKG@/libexec/lmod bash -t collection 2>&1 > /dev/null
-
-}
-
 _module() {
-	local cur="$2" prev="$3" cmds opts
-
+	local cur="${2}" prev="${3}" cmds opts
+	
 	COMPREPLY=()
-
+	
 	cmds="add avail delete help keyword list load purge rm restore save show spider swap \
 	      unload unuse update use whatis"
-
-	opts="-d -D -h -q -t -v -w  -s --style --expert --quiet --help  --quiet --terse --version --default --width -r --regexp --mt"
-
-	case "$prev" in
+	
+	opts="-d -D -h -q -t -v -w -s --style --expert --quiet --help --quiet --terse --version --default --width -r --regexp --mt"
+	
+	case "${prev}" in
 	add|load|try-load)
-                        COMPREPLY=( $(compgen -W "$(_module_not_yet_loaded)" -- "$cur") );;
-        restore)        COMPREPLY=( $(compgen -W "$(_module_savelist)" -- "$cur") );;
-        spider)         COMPREPLY=( $(compgen -W "$(_module_spider)" -- "$cur") );;
+		COMPREPLY=( $(compgen -W "$(_module_not_yet_loaded)" -- "${cur}") )
+		;;
 	rm|remove|unload|switch|swap)
-			COMPREPLY=( $(      compgen -W "$(_module_loaded_modules)" -- "$cur") );;
-	unuse)		COMPREPLY=( $(IFS=: compgen -W "${MODULEPATH}" -- "$cur") );;
-	use|*-a*)	_filedir -d || return 0;;
+		COMPREPLY=( $(compgen -W "$(_module_loaded_modules)" -- "${cur}") )
+		;;
+	restore)
+		COMPREPLY=( $(compgen -W "$(_module_savelist)" -- "${cur}") )
+		;;
+	spider)
+		COMPREPLY=( $(compgen -W "$(_module_spider)" -- "${cur}") )
+		;;
+	unuse)
+		COMPREPLY=( $(IFS=: compgen -W "${MODULEPATH}" -- "${cur}") )
+		;;
+	use|*-a*)
+		_filedir -d || return 0
```

Comment:
@pneerincx: We recently noticed that tab completion on `use` requires `_filedir`, which is not available by default in bash (it's not a builtin like `compgen` is), but is provided by a separate package (e.g. `bash-completion` on SL6)

Wouldn't it be better to check whether `_filedir` is defined rather than using it blindly?

I think that's what the `return 0` is about, but it doesn't do what it's intended to do.
Here's what I get on a system that doesn't have `_filedir` with `ml use /user<tab>`:

```
$ ml use /user-bash: _filedir: command not found
```